networkx/generators/degree_seq.py

```diff
@@ -699,13 +703,8 @@ def degree_sequence_tree(deg_sequence, create_using=None):
     # add the leaves
     for source in range(1, n - 1):
         nedges = deg.pop() - 2
-        for target in range(last, last + nedges):
-            G.add_edge(source, target)
+        G.add_edges_from((source, target) for target in range(last, last + nedges))
         last += nedges
-
-    # in case we added one too many
-    if len(G) > len(deg_sequence):
-        G.remove_node(0)
```

Comment:
Removing this is fine. I looked at the history and this is from before we even switched from cvs to svn. The code is pretty straightforward and I can't see any way we would get an extra node.

:)